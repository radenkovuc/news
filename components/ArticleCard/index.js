import styled from 'styled-components'
import React from 'react'
import withTranslation from '../HOCs/withTranslation'
import withContext from '../HOCs/withContext'
import {Link} from '../../i18n'
import {ARTICLE} from '../routes.json'

const Container = styled.div`
  width: 100%;
  border: 1px solid #444;
  border-radius: 10px;
  margin: 10px;
  padding: 15px;

  @media only screen and (min-width: 762px) {
    width: 300px;
  }
`

const Title = styled.div`
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
  text-align: left;
  font-family: 'Nunito Sans black', sans-serif;
  font-size: 18px;
  overflow: hidden;
  margin-bottom: 15px;
`

const Image = styled.div`
  width: 100%;
  height: 200px;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-image: ${(props) => `url('${props.url}')`};
`

const Description = styled.div`
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
  text-align: left;
  font-family: 'Nunito Sans black', sans-serif;
  font-size: 16px;
  overflow: hidden;
  margin-top: 15px;
`

const More = styled.div`
  display: flex;
  justify-content: flex-end;
  padding-top: 8px;
  cursor: pointer;
`
const MoreText = styled.div`
  font-family: 'Nunito Sans black', sans-serif;
  font-size: 18px;
  padding-right: 5px;
`
const MoreIcon = styled.div`
  width: 12px;
  height: 18px;
  background: url('/static/images/next.svg') no-repeat center/cover;
`

const ArticleCard = (props) => {
  const {article, appContext, t} = props

  const onClickMore = () => {
    appContext.setContextData({selectedArticle: article})
  }

  return (
    <Container>
      <Title>{article.title}</Title>
      <Image url={article.urlToImage} />
      <Description>{article.description}</Description>
      <Link href={ARTICLE.link}>
        <More onClick={onClickMore}>
          <MoreText>{t('MORE')}</MoreText>
          <MoreIcon />
        </More>
      </Link>
    </Container>
  )
}

export default withContext(withTranslation(ArticleCard))
