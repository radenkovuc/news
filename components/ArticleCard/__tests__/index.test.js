import React from 'react'
import ArticleCard from '../'
import renderer from 'react-test-renderer'

describe('<ArticleCard />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<ArticleCard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
