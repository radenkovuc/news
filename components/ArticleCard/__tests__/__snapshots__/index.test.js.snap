// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> match snapshot 1`] = `
<div
  className="Container-sc-4ap2jm-0 gvpCMM"
>
  <div
    className="Title-sc-4ap2jm-1 eFAftM"
  />
  <div
    className="Image-sc-4ap2jm-2 dMdvXD"
  />
  <div
    className="Description-sc-4ap2jm-3 McRdH"
  />
  <div
    className="More-sc-4ap2jm-4 dzZLkd"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="MoreText-sc-4ap2jm-5 jcFjZn"
    >
      MORE
    </div>
    <div
      className="MoreIcon-sc-4ap2jm-6 WmvRO"
    />
  </div>
</div>
`;
